tests/test_enterprise/views/test_grant_data_sharing_permissions.py

```diff
@@ -435,8 +463,22 @@ def test_post_course_specific_consent(
         assert resp.url.endswith(expected_redirect_url)  # pylint: disable=no-member
         assert resp.status_code == 302
         if not defer_creation:
+            dsc.refresh_from_db()
             assert dsc.granted is consent_provided
 
+        # we'll only create an enrollment record if (1) creation is not deferred,
+        # (2) consent is provided, and (3) we provide a course _run_ id
+        if (not defer_creation) and consent_provided and course_id.endswith('Demo_Course'):
+            mock_licensed_ent_course_enrollment_create.assert_called_once_with(
+                enterprise_course_enrollment=enterprise_enrollment,
+                license_uuid=license_uuid
```

Comment:
This feels a little weird to me? I'd expect we'd just check that a `LicensedEnterpriseCourseEnrollment` exists in the db?
Something like 
```python
assert LicensedEnterpriseCourseEnrollment.objects.get(
    enterprise_course_enrollment=enterprise_enrollment,
    license_uuid=license_uuid,
).exists() is True
```
maybe?

Reply:
agreed 👍